### PR TITLE
store/util: get the approximate number of records in region

### PR DESCRIPTION
**Pull request by AndreMouche**
Hi, 
     This PR has implemented the function to get the approximate number of records in the region, it's prepared for controlling the number of records in each region.

@huachaohuang @BusyJay PTAL

### CODE REVIEW

File: src/pd/pd.rs [hunk 1]
```diff
@@ -544,7 +544,12 @@ impl<T: PdClient> Runnable<Task> for Runner<T> {
             } => {
                 let approximate_size = match region_size {
                     Some(size) => size,
-                    None => get_region_approximate_size(&self.db, &region).unwrap_or(0),
+                    None => {
```
Comment (huachaohuang) [inline src/pd/pd.rs]: It's weird that you change this but don't use the result.
I think you can just report it to PD in this PR.

File: src/raftstore/coprocessor/split_check/size.rs [hunk 2]
```diff
@@ -81,7 +81,7 @@ impl<C: Sender<Msg> + Send> SplitCheckObserver for SizeCheckObserver<C> {
     fn add_checker(&self, ctx: &mut ObserverContext, host: &mut Host, engine: &DB) {
         let region = ctx.region();
         let region_id = region.get_id();
-        let region_size = match util::get_region_approximate_size(engine, region) {
+        let (_, region_size) = match util::get_region_approximate_stats(engine, region) {
```
Comment (huachaohuang) [inline src/raftstore/coprocessor/split_check/size.rs]: Not belong to this PR.

File: src/raftstore/coprocessor/split_check/size.rs [hunk 3]
```diff
@@ -147,7 +147,7 @@ mod tests {
     use raftstore::store::{keys, Msg, SplitCheckRunner, SplitCheckTask};
     use storage::ALL_CFS;
     use util::config::ReadableSize;
-    use util::properties::SizePropertiesCollectorFactory;
+    use util::properties::{MvccPropertiesCollectorFactory, SizePropertiesCollectorFactory};
```
Comment (huachaohuang) [inline src/raftstore/coprocessor/split_check/size.rs]: ditto

File: src/raftstore/coprocessor/split_check/size.rs [hunk 4]
```diff
@@ -160,6 +160,8 @@ mod tests {
         let path_str = path.path().to_str().unwrap();
         let db_opts = DBOptions::new();
         let mut cf_opts = ColumnFamilyOptions::new();
+        let f = Box::new(MvccPropertiesCollectorFactory::default());
```
Comment (huachaohuang) [inline src/raftstore/coprocessor/split_check/size.rs]: ditto

File: src/raftstore/store/util.rs [hunk 5]
```diff
@@ -229,6 +229,28 @@ pub fn get_region_approximate_size(db: &DB, region: &metapb::Region) -> Result<u
...
+    let collection = db.get_properties_of_tables_in_range(cf, &[range])?;
+    for (_, v) in &*collection {
+        let props = RowsProperties::decode(v.user_collected_properties())?;
+        count += props.get_approximate_rows_in_range(&start, &end);
+    }
+    Ok(count)
+}
+
+/// Get the approximate number of records and size in the region.
+pub fn get_region_approximate_stats(db: &DB, region: &metapb::Region) -> Result<(u64, u64)> {
```
Comment (siddontang) [inline src/raftstore/store/util.rs]: I prefer defining a Stat structure here instead of the tuple, maybe we may support more stats later.

File: src/raftstore/store/util.rs [hunk 6]
```diff
@@ -229,6 +229,36 @@ pub fn get_region_approximate_size(db: &DB, region: &metapb::Region) -> Result<u
     Ok(size)
 }
 
+/// Get the approximate number of records in the region.
+pub fn get_region_approximate_count(db: &DB, region: &metapb::Region) -> Result<u64> {
```
Comment (huachaohuang) [inline src/raftstore/store/util.rs]: I think `get_region_approximate_rows` is more clear.

File: src/raftstore/store/util.rs [hunk 7]
```diff
@@ -229,6 +229,36 @@ pub fn get_region_approximate_size(db: &DB, region: &metapb::Region) -> Result<u
...
+    for (_, v) in &*collection {
+        let props = RowsProperties::decode(v.user_collected_properties())?;
+        count += props.get_approximate_rows_in_range(&start, &end);
+    }
+    Ok(count)
+}
+
+pub struct RegionApproximateStat {
+    /// the approximate number of records in the region.
+    pub count: u64,
```
Comment (huachaohuang) [inline src/raftstore/store/util.rs]: I think `count` means a lot of things, I think `rows` is more clear.

**Comment by AndreMouche** [comment]
@zhangjinpeng1987 PTAL
